li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -349,4 +365,27 @@ public void close(Duration timeout) {
   public void wakeup() {
     throw new UnsupportedOperationException("Not implemented yet");
   }
+
+  public LiKafkaConsumer<K, V> getPerClusterConsumer(ClusterDescriptor cluster) {
```

Comment:
does this need to be public ? we're trying to "fake" a simple client

Reply:
This is used for unit tests. I can add a comment saying that it is intended for testing only. Would that be acceptable?

Reply:
Per our discussion, changed it to package private. Also added a comment.